src/concurrency.md

```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
 }
 ```
 
-The example uses `crossbeam` crate, which provides data structures and functions 
+The example uses `crossbeam` crate, which provides data structures and functions
 for concurrent and parallel programming. [`Scope::spawn`] spawns a new scoped thread that is guaranteed
 to terminate before returning from the closure that passed into [`crossbeam::scope`] function, meaning that
 you can reference data from the calling function.
 
+[ex-threadpool-fractal]: #ex-threadpool-fractal
+<a name="ex-threadpool-fractal"></a>
+## Draw fractal dispatching work to a thread pool
+
+[![threadpool-badge]][threadpool] [![num-badge]][num] [![num_cpus-badge]][num_cpus] [![image-badge]][image] [![cat-concurrency-badge]][cat-concurrency][![cat-science-badge]][cat-science][![cat-rendering-badge]][cat-rendering]
+
+Draws a fractal from [Julia set] to an image utilizing a thread pool for computation.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate threadpool;
+extern crate num;
+extern crate num_cpus;
+extern crate image;
+
+use std::sync::mpsc::{channel, RecvError};
+use threadpool::ThreadPool;
+use num::complex::Complex;
+use image::{ImageBuffer, Pixel, Rgb};
+#
+# error_chain! {
+#     foreign_links {
+#         MpscRecv(RecvError);
+#     }
+# }
+
+// Function converting intensity values to RGB
+// Based on http://www.efg2.com/Lab/ScienceAndEngineering/Spectra.htm
+fn wavelength_to_rgb(wavelength: u32) -> Rgb<u8> {
```

Comment:
Hide this like we hide all the error handling code. This is not important to understanding threadpool.